### PR TITLE
Size type fixes

### PR DESCRIPTION
**Pull request by StelioKontosXBL**
Migrating `X<T>::size_type` from `int` to `size_t` for consistency. Also fixes warnings when comparing signed/unsigned values, removing the need for additional unnecessary casting in most situations.

### CODE REVIEW

File: intrusive_list.h [hunk 1]
```diff
@@ -76,9 +76,11 @@ class intrusive_list_base
 	void pop_back()		{ unlink(m_root.prev); }
 	void pop_front()	{ unlink(m_root.next); }
 
+	// NOTE: commenting `intrusive_list_base::size()` method declaration; missing definition ~SK
 	// @note: allow for constant complexity way of checking this
 	// (at a cost of additional variable)?
-	size_type size() const;
+	//size_type size() const;
```
Comment (StelioKontosXBL) [inline intrusive_list.h:82]: @msinilo is this declaration serving as a concept or "template guide" for classes deriving from it? If so, I'll revert this change. See also the same instance in intrusive_slist.h.

File: intrusive_slist.h [hunk 2]
```diff
@@ -62,14 +62,18 @@ class intrusive_slist_base
...
 	// @note: allow for constant complexity way of checking this
 	// (at a cost of additional variable)?
-	size_type size() const;
+	//size_type size() const;
+
 	bool empty() const	{ return !m_root.in_list(); }
 
 protected:
+	intrusive_slist_base();
+
```
Comment (StelioKontosXBL) [inline intrusive_slist.h:76]: Changed visibility of the ctor declaration to protected, to match how intrusive_list.h had it.

**Comment by StelioKontosXBL** [comment]
Changes are all complete at this point; I'm leaving this pr as a draft pending running the tests with these changes, though. I expect to be able to get to that before the end of the weekend.

File: intrusive_list.h (hunk 1 above)
Comment (msinilo) [reply intrusive_list.h:82]: Definition is in the .cpp file (base is not a template)

Comment (StelioKontosXBL) [reply intrusive_list.h:82]: Well, this is kind of awkward...reverting noob mistake lol. Thanks.

File: intrusive_slist.h (hunk 2 above)
Comment (StelioKontosXBL) [reply intrusive_slist.h]: This change was reverted with the others in this commit.

**Pull request merged**